active_learning_multiannotator/active_learning_single_annotator.ipynb

```diff
@@ -0,0 +1,461 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset = ImageDataset.from_folder('./256_ObjectCategories/')\n",
+    "dataset = dataset.replace(257, 256) # no class class in dataset is labeled as 257, we need to reindex"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get train-test split\n",
+    "X, y = np.arange(len(dataset) * 2).reshape((len(dataset), 2)), range(len(dataset))\n",
+    "_, _, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=123)\n",
+    "\n",
+    "df_train = dataset.iloc[y_train]\n",
+    "df_test = dataset.iloc[y_test]\n",
+    "df_test = df_test.reset_index(drop=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df_train, df_test = train_test_split(dataset, test_size=0.33, random_state=123)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_labeled(dataset,  num_labeled_per_class=15):\n",
+    "    \"\"\"Splits provided dataset into two datasets. With df_labeled containing num_labeled_per_class labeles for \n",
+    "    each class and df_unlabeled containing the rest of the rows in dataset\"\"\"\n",
+    "    \n",
+    "    df_labeled = dataset.groupby(\"label\").sample(n=num_labeled_per_class, random_state=123)\n",
+    "    labeled_index = list(df_labeled.index)\n",
+    "    unlabeled_index = [i for i in range(len(dataset)) if i not in labeled_index]\n",
+    "    df_unlabeled = dataset.iloc[unlabeled_index]\n",
+    "    df_unlabeled = df_unlabeled.reset_index(drop=True)\n",
+    "    df_labeled = df_labeled.reset_index(drop=True)    \n",
+    "    return labeled_index, df_labeled, unlabeled_index, df_unlabeled\n",
+    "\n",
+    "labeled_index, df_labeled, unlabeled_index, df_unlabeled = get_labeled(dataset, num_labeled_per_class=8)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Train model to obtain predicted probabilites\n",
+    "\n",
+    "First, we train our model on a set of labels obtained by `get_labeled` to get the out-of-sample predicted class probabilities for both the labeled and unlabeled data. \n",
+    "\n",
+    "The train function will return two sets of predicted probabilites, `pred_probs_labeled` are the predicted probabilites for examples that have existing annotator labels (they correspond directly with the rows in `df_labeled`), whereas `pred_probs_unlabeled` are the predicted probabilites for examples that do not have any annotator labels (they correspond directly with the rows in `df_unlabeled`). These predicted probabilities will later be used to compute the active learning score.\n",
```

Comment:
why are you still returning pred_probs_labeled? this does not seem necessary when we are doing active learning with only one label per example. One would never be interested in scoring the already-labeled data in this case.